araproc/analysis/correlation.py

```diff
@@ -0,0 +1,114 @@
+import ROOT
+import numpy as np
+import math
+import os
+
+# HELP HERE: constants that we might want to store elsewhere, or can I retrieve them somehow?
+channel_num = 16
+
+def get_surface_correlation_ratio(corr_map, station_id, z_thresh):
+    """
+    Calculates the ratio of the maximum surface correlation value within a specific
+    theta range to the overall peak correlation value. The theta range is determined
+    based on the average antenna z-coordinate and a threshold.
+
+    Parameters
+    ----------
+    corr_map : dict
+        A dictionary containing:
+        - 'map': ROOT.TH2D histogram with correlation values mapped by theta and phi.
+        - 'corr': The overall peak correlation value.
+        - 'radius': The correlation radius for the station.
+    station_id : int
+        The station ID for which to calculate the correlation ratio.
+    z_thresh : float
+        The threshold value to define the lower bound of the theta range.
+
+    Returns
+    -------
+    surf_corr_ratio : float
+        The ratio of the maximum surface correlation value within the theta range 
+        to the overall peak correlation.
+    max_surf_theta : float
+        The theta angle corresponding to the maximum surface correlation value.
+    max_surf_phi : float
+        The phi angle corresponding to the maximum surface correlation value.
+    """
+    
+    # Extract the radius from corr_map
+    corr_radius = float(corr_map.get("radius", None))
```

Comment:
Is this the radius of the global maximum? Does this assume a specific map is passed?

Reply:
Yeah, it's the radius of the given map. `corr_map` should be something like `reco_results["distant_v"]` and later we'll catch if the map was created at a large enough radius.

Reply:
I wonder if we need pass multiple maps? eg what if the max correlation occurs in a different map than the max surface correlation?

Reply:
You'd calculate the surface correlation ratio for that specific map, right? At this point, both the global max and surface corr max are calculated on the same map.

Reply:
I'm not sure. For example if the global max across all maps is at the calpulser radius, we still would be interested in knowing the surface correlation ratio. Right now that's not defined. I think we want the surface correlation ratio to be (max surface correlation over all maps)/(max correlation over all maps). Maybe there's a subtlety about the polarizations, but ignoring that for a second I think that's what we want. We can talk about it on the call if you want.

Reply:
Okay, I can do whatever needed. Just give the order haha. I did it how I understood. 

I will push all the other changes now, including moving this function to standard_reco.py